File: 2021/09/05/main.go
package main

import "sort"

func orderlyQueue(s string, k int) string {
	n := len(s)
	if n == 1 {
		return s
	}
	if k > 1 {
		return sorted(s)
	}
	s2 := s + s
	r := "z"
	for i := 1; i < n; i++ {
		r += "z"
	}
	for i := 0; i < n; i++ {
		t := s2[i : i+n]
		if t < r {
			r = t
		}
	}
	return r
}

func sorted(s string) string {
	bytes := []byte(s)
	sort.Slice(bytes, func(i int, j int) bool {
		return bytes[i] < bytes[j]
	})
	return string(bytes)
}

File: 2021/09/05/main.py
class Solution:
    def orderlyQueue(self, s: str, k: int) -> str:
        n = len(s)
        if n == 1:
            return s
        if k > 1:
            return ''.join(sorted(s))
        s2 = 2 * s
        r = "z" * n
        for i in range(n):
            t = s2[i: i + n]
            if t < r:
                r = t
        return r